cogs/events.py

```diff
@@ -1,148 +1,132 @@
+        # Restore the app commands error handler
+        self.bot.tree.on_error = self.old_error_handler
+
+    async def on_app_command_error(
+        self, interaction: discord.Interaction, error: Union[app_commands.AppCommandError, Exception]
+    ) -> Any:
+        """The event triggered when an error is raised while invoking an app command."""
+        TO_IGNORE = (app_commands.CommandNotFound,)
+        SEND_MESSAGES = (app_commands.MissingPermissions, app_commands.BotMissingPermissions, NotOwner, NoPrivilege)
+        if isinstance(error, app_commands.CommandInvokeError):
+            error = error.original
+
+        if isinstance(error, TO_IGNORE):
+            return
+        elif isinstance(error, SEND_MESSAGES):
+            return await interaction.response.send_message(str(error), ephemeral=True)
+        elif isinstance(error, app_commands.CommandOnCooldown):
+            await interaction.response.send_message(
+                "This command is on cooldown... " f"try again in {error.retry_after:.2f} seconds.", ephemeral=True
             )
-
-        elif isinstance(err, errors.CommandOnCooldown):
-            await ctx.response.send_message(
-                "This command is on cooldown... "
-                f"try again in {err.retry_after:.2f} seconds."
+        else:
+            # Better to send to a dev channel...
+            trace = default.traceback_maker(error)
+            await interaction.response.send_message(
+                f"There was an error processing the command ;-;\n{trace}", ephemeral=True
             )
 
-        elif isinstance(err, errors.CommandNotFound):
-            pass
-
     @commands.Cog.listener()
     async def on_guild_join(self, guild: discord.Guild):
-        to_send = next((
-            chan for chan in guild.text_channels
-            if chan.permissions_for(guild.me).send_messages
-        ), None)
+        to_send = next((chan for chan in guild.text_channels if chan.permissions_for(guild.me).send_messages), None)
 
         if to_send:
-            await to_send.response.send_message(
-                self.bot.config.discord_join_message
-            )
+            await to_send.send(self.bot.config.discord_join_message)
 
     @commands.Cog.listener()
-    async def on_command(self, ctx: discord.Interaction):
-        location_name = ctx.guild.name if ctx.guild else "Private message"
-        print(f"{location_name} > {ctx.user} > {ctx.message.clean_content}")
+    async def on_app_command_completion(self, interaction: discord.Interaction) -> None:
+        if not interaction.command:  # won't happen, just to shut the linter
+            return
+
+        location_name = interaction.guild.name if interaction.guild else "Private message"
+        print(
+            f"{location_name} > {interaction.user} > {interaction.command.qualified_name} (arguments: {vars(interaction.namespace)})"
+        )
 
     @commands.Cog.listener()
     async def on_ready(self):
```

Comment:
Doing any of this in on_ready really sets a bad example. This needs to go in setup_hook.